### DISCUSSION TITLE
Public MQTT access

### DISCUSSION THREAD

**Discussion by JonasSL** [teslamate-org/teslamate#2700] [Q&A] [answered]
Hi,

My Teslamate instance is running on an external network with Let's Encrypt and Traefik, but I also want to access the MQTT brokers provided. 
I have another MQTT broker located elsewhere, and I want to make a MQTT bridge between the two. 

Is that possible in a secure way? And if so - how?
I'm not that experienced in docker-compose/traefik setup, so that's why I'm hoping to get some help here.

Thanks!
/Jonas

**Comment by tobiasehlert** [comment, marked as answer]
Hi @JonasSL,

You can tell TeslaMate to run towards another MQTT somewhere else, eg. a mosquitto broker located on another server or on a VPS in some cloud provider.

You'd just have to configure TeslaMate towards that instance by changing the MQTT environmental variables (which can be found [here](https://docs.teslamate.org/docs/configuration/environment_variables)) and you should be ready to go.
Since you use docker-compose with Traefik, you should already have some MQTT parameter in there pointing to the other docker container already. Change that value and add more ones like _MQTT_TLS_ and more and you should get it to work.

Regarding the security concern, try to have the access locked down as much as possible, so not any outside person could connect to it. Also use TLS and authentication on top of that :)

Regards,
Tobias

**Reply by JonasSL** [reply]
Thank you for your answer!

That's is a good idea - I'll definitely look into that.
